sdk/typescript/test/e2e/local-txn-builder.test.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, beforeAll } from 'vitest';
+import {
+  getExecutionStatusType,
+  LocalTxnDataSerializer,
+  RawSigner,
+} from '../../src';
+import {
+  DEFAULT_RECIPIENT,
+  DEFAULT_GAS_BUDGET,
+  setup,
+  TestToolbox,
+} from './utils/setup';
+
+describe('Local Transaction Builder', () => {
+  let toolbox: TestToolbox;
+  let signer: RawSigner;
+
+  beforeAll(async () => {
+    toolbox = await setup('fullnode');
+    signer = new RawSigner(
+      toolbox.keypair,
+      toolbox.provider,
+      new LocalTxnDataSerializer(toolbox.provider)
+    );
+  });
+
+  it('Split coin', async () => {
+    const coins = await toolbox.provider.getGasObjectsOwnedByAddress(
+      toolbox.address()
+    );
+    const txn = await signer.splitCoinWithRequestType({
+      coinObjectId: coins[0].objectId,
+      splitAmounts: [DEFAULT_GAS_BUDGET * 2],
+      gasBudget: DEFAULT_GAS_BUDGET,
+      gasPayment: coins[1].objectId,
+    });
+    expect(getExecutionStatusType(txn)).toEqual('success');
```

Comment:
You could also actually validate this does what you expect, by ensuring there are two tokens. But that's veering closer to testing RPC itself.